### ISSUE TITLE
When i compile and run the tests I receive the following exception

### ISSUE THREAD

**Issue by nokeechia** [unicredit/hbase-rdd#21]
An exception or error caused a run to abort: ROLLINGUPGRADE 
java.lang.NoSuchFieldError: ROLLINGUPGRADE
	at org.apache.hadoop.hdfs.MiniDFSCluster.createArgs(MiniDFSCluster.java:1080)
	at org.apache.hadoop.hdfs.MiniDFSCluster.createNameNode(MiniDFSCluster.java:1106)
	at org.apache.hadoop.hdfs.MiniDFSCluster.createNameNodesAndSetConf(MiniDFSCluster.java:978)
	at org.apache.hadoop.hdfs.MiniDFSCluster.initMiniDFSCluster(MiniDFSCluster.java:807)
	at org.apache.hadoop.hdfs.MiniDFSCluster.<init>(MiniDFSCluster.java:738)
	at org.apache.hadoop.hbase.HBaseTestingUtility.startMiniDFSCluster(HBaseTestingUtility.java:558)
	at org.apache.hadoop.hbase.HBaseTestingUtility.startMiniCluster(HBaseTestingUtility.java:937)
	at org.apache.hadoop.hbase.HBaseTestingUtility.startMiniCluster(HBaseTestingUtility.java:819)
	at org.apache.hadoop.hbase.HBaseTestingUtility.startMiniCluster(HBaseTestingUtility.java:813)
	at org.apache.hadoop.hbase.HBaseTestingUtility.startMiniCluster(HBaseTestingUtility.java:757)
	at unicredit.spark.hbase.MiniCluster$class.beforeAll(MiniCluster.scala:26)
	at unicredit.spark.hbase.WriteBulkSpec.beforeAll(WriteBulkSpec.scala:9)
	at org.scalatest.BeforeAndAfterAll$class.beforeAll(BeforeAndAfterAll.scala:187)
	at unicredit.spark.hbase.WriteBulkSpec.beforeAll(WriteBulkSpec.scala:9)
	at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:253)
	at unicredit.spark.hbase.WriteBulkSpec.org$scalatest$BeforeAndAfter$$super$run(WriteBulkSpec.scala:9)
	at org.scalatest.BeforeAndAfter$class.run(BeforeAndAfter.scala:241)
	at unicredit.spark.hbase.WriteBulkSpec.run(WriteBulkSpec.scala:9)
	at org.scalatest.tools.SuiteRunner.run(SuiteRunner.scala:55)
	at org.scalatest.tools.Runner$$anonfun$doRunRunRunDaDoRunRun$3.apply(Runner.scala:2563)
	at org.scalatest.tools.Runner$$anonfun$doRunRunRunDaDoRunRun$3.apply(Runner.scala:2557)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at org.scalatest.tools.Runner$.doRunRunRunDaDoRunRun(Runner.scala:2557)
	at org.scalatest.tools.Runner$$anonfun$runOptionallyWithPassFailReporter$2.apply(Runner.scala:1044)
	at org.scalatest.tools.Runner$$anonfun$runOptionallyWithPassFailReporter$2.apply(Runner.scala:1043)
	at org.scalatest.tools.Runner$.withClassLoaderAndDispatchReporter(Runner.scala:2722)
	at org.scalatest.tools.Runner$.runOptionallyWithPassFailReporter(Runner.scala:1043)
	at org.scalatest.tools.Runner$.run(Runner.scala:883)
	at org.scalatest.tools.Runner.run(Runner.scala)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.runScalaTest2(ScalaTestRunner.java:138)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.main(ScalaTestRunner.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)

This seems to be associated with the initialization of the MiniCluster.

Have you got any idea?

**Comment by fralken** [comment]
I confirm that running scalatests from intellij you get this error, I don't know why.

But if you run tests from sbt (i.e. run `sbt test` from a console)  they should work

**Comment by mfirry** [comment]
Could it be related to how hdfs gets started-up for testing purposes? Seems somehow to refer to [StartupOption.ROLLINGUPGRADE](https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/common/HdfsServerConstants.java#L152) enum value.

**Comment by fralken** [comment]
I investigated further, and I figured out that it is a matter of dependency resolution.
For example spark-core 1.5.0 depends on hadoop-hdfs 2.2.0 while hadoop-common 2.6.0-cdh5.5.1 depends on hadoop-hdfs 2.6.0cdh5.5.1.

It appears that when scalatests are executed via IntelliJ, hadoop-hdfs 2.2.0 is loaded and we get NoSuchFieldError, while when executed via sbt hadoop-hdfs 2.6.0cdh5.5.1 is loaded and everything is ok. It is not a fault of cdh, but rather the mix of spark 1.5.0 with hadoop 2.6.0.

I played around with versions, but with any combination the problem moves somewhere else.
For example, if I set spark version to 1.5.0cdh5.5.1, it correctly depends on hadoop-hdfs 2.6.0cdh5.5.1, but then there is a conflict with com.google.guava, and even the compilation fails...

Welcome to dependency hell...
